### DISCUSSION TITLE
Code Deserialization Error when publishing

### DISCUSSION THREAD

**Discussion by ethan-douglass** [aptos-labs/aptos-developer-discussions#546] [Questions]
### Discord user ID

ThatGoatHerder

### Describe your question in detail.

Hello!

I am really struggling to learn the move CLI(I've done solidity for multiple years and wanted to give it a shot)

When I try using one of the example projects and follow the instructions, I am hit with a code deserialization error when I got to publish. 

Both aptos move test, and aptos move compile pass successfully. 

What am I doing wrong?

### What error, if any, are you getting?

Error": "Simulation failed with status: CODE_DESERIALIZATION_ERROR\nExecution failed with message: Backtrace: Backtrace [\n    { fn: \"move_binary_format::errors::PartialVMError::finish\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::deserialize_module_bundle\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::resolve_pending_code_publish\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::execute_script_or_entry_function\" },\n    { fn: \"aptos_vm::aptos_vm::AptosSimulationVM::create_vm_and_simulate_signed_transaction\" },\n    { fn: \"aptos_api::transactions::TransactionsApi::simulate\" },\n    { fn: \"aptos_api::transactions::TransactionsApi::simulate_transaction::{{closure}}::{{closure}}\" },\n    { fn: \"<tracing::instrument::Instrumented<T> as core::future::future::Future>::poll\" },\n    { fn: \"tokio::runtime::task::core::Core<T,S>::poll\" },\n    { fn: \"tokio::runtime::task::harness::Harness<T,S>::poll\" },\n    { fn: \"tokio::runtime::blocking::pool::Inner::run\" },\n    { fn: \"std::sys_common::backtrace::__rust_begin_short_backtrace\" },\n    { fn: \"core::ops::function::FnOnce::call_once{{vtable.shim}}\" },\n    { fn: \"std::sys::pal::unix::thread::Thread::new::thread_start\" },\n    { fn: \"start_thread\" },\n    { fn: \"clone\" },\n]"

### What have you tried or looked at?  Or how can we reproduce the error?

Ive tried uninstalling and reinstalling
aptos move clean
trying different example projects from different places
I've tried on a different computer as well
nothing has worked

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

### Describe your environment or tooling in detail

Macos Sequoia 15.2 
Aptos CLI 4.6(newest release that just came out)

**Comment by rahxephon89** [comment]
Hi @ethan-douglass, could you provide the CLI version and the concrete command you were using? Thanks

**Reply by ethan-douglass** [reply]
CLI is 4.6.0
Below is the full output when I run aptos move publish.
`aptos move publish
Compiling, may take a little while to download git dependencies...
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-framework.git
INCLUDING DEPENDENCY AptosFramework
INCLUDING DEPENDENCY AptosStdlib
INCLUDING DEPENDENCY MoveStdlib
BUILDING HelloBlockchainExample
package size 1803 bytes
{
  "Error": "Simulation failed with status: CODE_DESERIALIZATION_ERROR\nExecution failed with message: Backtrace: Backtrace [\n    { fn: \"move_binary_format::errors::PartialVMError::finish\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::deserialize_module_bundle\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::resolve_pending_code_publish\" },\n    { fn: \"aptos_vm::aptos_vm::AptosVM::execute_script_or_entry_function\" },\n    { fn: \"aptos_vm::aptos_vm::AptosSimulationVM::create_vm_and_simulate_signed_transaction\" },\n    { fn: \"aptos_api::transactions::TransactionsApi::simulate\" },\n    { fn: \"aptos_api::transactions::TransactionsApi::simulate_transaction::{{closure}}::{{closure}}\" },\n    { fn: \"<tracing::instrument::Instrumented<T> as core::future::future::Future>::poll\" },\n    { fn: \"tokio::runtime::task::core::Core<T,S>::poll\" },\n    { fn: \"tokio::runtime::task::harness::Harness<T,S>::poll\" },\n    { fn: \"tokio::runtime::blocking::pool::Inner::run\" },\n    { fn: \"std::sys_common::backtrace::__rust_begin_short_backtrace\" },\n    { fn: \"core::ops::function::FnOnce::call_once{{vtable.shim}}\" },\n    { fn: \"std::sys::pal::unix::thread::Thread::new::thread_start\" },\n    { fn: \"start_thread\" },\n    { fn: \"clone\" },\n]"
}
`

**Reply by rahxephon89** [reply]
I could not reproduce the issue when publishing the hello_blockchain project in `aptos-core/aptos-move/move-example` using the latest CLI 4.6.0 installed by `brew install aptos`. Could you share the package that failed? And have you succeeded in publishing any other packages?

**Comment by D4mph1r** [comment]
i am also facing the same issue and i am using the cli aptos 7.3.0.
